### PR TITLE
OCLOMRS-892: Reset the Error Message on create and edit dictionary form

### PR DESCRIPTION
**Pull request by jwnasambu**
# JIRA TICKET NAME:
https://issues.openmrs.org/browse/OCLOMRS-892

# Summary:
I added a react hook to reset the form state when you recreate a form.

### CODE REVIEW

File: src/apps/dictionaries/pages/CreateDictionaryPage.tsx [hunk 1]
```diff
@@ -26,6 +27,7 @@ interface Props {
   ) => void;
   loading: boolean;
   newDictionary?: APIDictionary;
+  resetCreateDictionaryAction: () => void;
```
Comment (ibacher) [inline src/apps/dictionaries/pages/CreateDictionaryPage.tsx:30]:
```suggestion
  resetCreateDictionary: () => void;
```

Comment (ibacher) [reply src/apps/dictionaries/pages/CreateDictionaryPage.tsx:30]: The name in the `Props` interface needs to match the name on line 39 and the name in `mapActionsToProps`.

File: src/apps/dictionaries/pages/EditDictionaryPage.tsx [hunk 2]
```diff
@@ -96,6 +97,8 @@ const EditDictionaryPage: React.FC<Props> = ({
     linkedSource,
     createAndAddLinkedSource
   ]);
+  
+  useEffect(() => resetEditDictionary, []);
```
Comment (ibacher) [inline src/apps/dictionaries/pages/EditDictionaryPage.tsx:101]: `resetEditDictionary` also needs to be added to `ActionProps` and the start of the `EditDictionaryPage` function.

**Review by ibacher — CHANGES_REQUESTED** [review]

File: src/apps/dictionaries/pages/CreateDictionaryPage.tsx [hunk 3]
```diff
@@ -34,9 +36,13 @@ const CreateDictionaryPage: React.FC<Props> = ({
   errors,
   createSourceAndDictionary,
   loading,
+  resetCreateDictionary,
   newDictionary
 }: Props) => {
   const previouslyLoading = usePrevious(loading);
+  
+  // eslint-disable-next-line react-hooks/exhaustive-dep
```
Comment (ibacher) [inline src/apps/dictionaries/pages/CreateDictionaryPage.tsx:44]:
```suggestion
  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Linter rule names are very specific!

**Comment by jwnasambu** [comment]
@ibacher Thanks so for much for reviewing my work.

**Review by ibacher — APPROVED** [review]

**Pull request merged**